registers.go

```diff
@@ -19,7 +19,7 @@ func (r *reg) set(offset, val uint8) bool {
 		tmpVal := uint8((*r) << 4 >> 4)
 		*r = reg(tmpVal | (val << 4))
 	} else {
-		isZero = *r > 15
```

Comment:
when `*r == 17` (0x11), `isZero` would be `true` and it cause `nz` decreasing.
this is the problem.